lib/tasks/default.rake

```diff
@@ -373,57 +373,74 @@ task :save_en do
   sh 'cp -p config/locales/en.yml config/locales/en.yml.ORIG'
 end
 
+desc 'Restore English translation file from .ORIG file'
+task :restore_en do
+  sh 'mv config/locales/en.yml.ORIG config/locales/en.yml'
+end
+
 desc 'Fix locale text'
 task :fix_locale do
   # Google Translate generates text that has predictable errors.
   # This task fixes some predictable errors automatically.
   puts "\nFixing locale text.\n"
-  # Rake won't let us define global constants, so we'll do what we can
-  locale_files = './config/locales/localization*.yml ' \
-                 './config/locales/translation*.yml'
-  sh %q{ruby -pi -e "sub(/< a /, '<a ')" \
-        -e "sub(/< \057/, '</')" -e "sub(/<\057 /, '</')" \
-        -e "sub(/<Strong>/, '<strong>')" -e "sub(/<Em>/, '<em>')" \
-        -e "sub(/ Href *=/, 'href=')" \
-        -e "sub(/href = /, 'href=')" \
-        -e "sub(/class = /, 'class=')" \
-        -e "sub(/target = /, 'target=')" } + locale_files
-  # Show whether or not there are known problems after it.
+  filenames = Dir[Rails.root.join('config', 'locales', 'translation.*.yml')]
+  filenames.each do |filename|
+    IO.write(
+      filename, IO.read(filename)
+                  .gsub(/< a /, '<a ')
+                  .gsub(/< \057/, '</')
+                  .gsub(/<\057 /, '</')
+                  .gsub(/<Strong>/, '<strong>')
+                  .gsub(/<Em>/, '<em>')
+                  .gsub(/ Href *=/, 'href=')
+                  .gsub(/href = /, 'href=')
+                  .gsub(/class = /, 'class=')
+                  .gsub(/target = /, 'target=')
+    )
+  end
 end
 
 # Test locale files; returns true if successful (no errors)
-def test_locale_files
+def locale_files_compliant?
+  # Check text.  Use 'system' so we can get the success result.
   system 'rails test test/models/translations_test.rb'
 end
 
-# Fix up translation:sync.
-# First, translation:sync rewrites the source en.yml file, which it shouldn't
-# ever do, and in the process reformats it into garbage with overly-long lines.
-# We modify its behavior to save the en.yml file, and later restore it.
-# We also remove trailing whitespace after running "translation:sync".
+require 'yaml'
+desc 'Fix translation whitespace'
+task :fix_i18n_whitespace do
+  # Reformat YAML with a line-wdith of 80, which also gets rid of the
+  # white space at the end of quotes and switches to the more readable
+  # YAML folding style. Separately, delete whitespace as the end of
+  # lines.
+  filenames = Dir[Rails.root.join('config', 'locales', 'translation.*.yml')]
+  filenames.each do |filename|
+    IO.write(
+      filename, YAML.load_file(filename).to_yaml(line_width: 80)
+                    .gsub(/ $/, '')
+    )
```

Comment:
This has nothing to do with the code here, but we might want to do the same thing once to en.yml in a separate one-time "out of bounds" conversion.

I didn't realize that translation.io was reformatting our text into such an ugly form until it became hard to switch manually.